app/code/Magento/NewsletterGraphQl/Model/Resolver/SubscribeEmailToNewsletter.php

```diff
@@ -0,0 +1,246 @@
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Framework\Validator\EmailAddress as EmailValidator;
+use Magento\Newsletter\Model\ResourceModel\Subscriber as SubscriberResourceModel;
+use Magento\Newsletter\Model\Subscriber;
+use Magento\Newsletter\Model\SubscriptionManagerInterface;
+use Magento\Store\Model\ScopeInterface;
+
+/**
+ * Resolver class for the `subscribeEmailToNewsletter` mutation
+ */
+class SubscribeEmailToNewsletter implements ResolverInterface
+{
+    /**
+     * @var CustomerAccountManagement
+     */
+    private $customerAccountManagement;
+
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @var EmailValidator
+     */
+    private $emailValidator;
+
+    /**
+     * @var EnumLookup
+     */
+    private $enumLookup;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var SubscriberResourceModel
+     */
+    private $subscriberResource;
+
+    /**
+     * @var SubscriptionManagerInterface
+     */
+    private $subscriptionManager;
+
+    /**
+     * SubscribeEmailToNewsletter constructor.
+     *
+     * @param CustomerAccountManagement $customerAccountManagement
+     * @param CustomerRepositoryInterface $customerRepository
+     * @param EmailValidator $emailValidator
+     * @param EnumLookup $enumLookup
+     * @param ScopeConfigInterface $scopeConfig
+     * @param SubscriberResourceModel $subscriberResource
+     * @param SubscriptionManagerInterface $subscriptionManager
+     */
+    public function __construct(
+        CustomerAccountManagement $customerAccountManagement,
+        CustomerRepositoryInterface $customerRepository,
+        EmailValidator $emailValidator,
+        EnumLookup $enumLookup,
+        ScopeConfigInterface $scopeConfig,
+        SubscriberResourceModel $subscriberResource,
+        SubscriptionManagerInterface $subscriptionManager
+    ) {
+        $this->customerAccountManagement = $customerAccountManagement;
+        $this->customerRepository = $customerRepository;
+        $this->emailValidator = $emailValidator;
+        $this->enumLookup = $enumLookup;
+        $this->scopeConfig = $scopeConfig;
+        $this->subscriberResource = $subscriberResource;
+        $this->subscriptionManager = $subscriptionManager;
+    }
+
+    /**
+     * Adds an email into a newsletter subscription
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     * @return array|Value|mixed
+     * @throws GraphQlInputException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $email = trim($args['email']);
+
+        if (empty($email)) {
+            throw new GraphQlInputException(__('Specify the "email" value.'));
```

Comment:
I understand that you might not be able to change exception messages, because they already exist elsewhere.

```suggestion
            throw new GraphQlInputException(__('You must specify an email address to subscribe to a newsletter.'));
```

Reply:
**implemented**

Reply:
> I understand that you might not be able to change exception messages, because they already exist elsewhere.

Awesome suggestions, @keharper !

I've implemented all your suggestions! 

Thank you !!!